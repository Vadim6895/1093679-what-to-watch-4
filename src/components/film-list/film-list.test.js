import React from "react";
import renderer from "react-test-renderer";
import FilmList from "./film-list.jsx";

const filmCards = [{
  movieName: `The Grand Budapest Hotel`,
  productionDate: `1984`,
  genre: `Drama`,
  moviePoster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  moviePreview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  id: 0,
  director: `Anthony Mann`,
  actors: [`Anthony Mann`],
  rating: 9,
  ratingsQuantity: 250,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.Lorem ipsum dolor sit amet, consectetur adipiscing elit.Lorem ipsum .`,
  length: 120,
  backgroundColor: `#ffffff`,
  backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  isFavorite: false,
  reviews: [],
}];
const CARDS_COUNT = 8;

it(`Should FilmList render correctly`, () => {
  const tree = renderer
  .create(<FilmList
    onFilmClick={() => {}}
    renderPlayer={() => {}}
    actualCardsCount={CARDS_COUNT}
    actualCards={filmCards.slice(0, 8)}
    newCards={filmCards.slice(0, 8)}
    onClick={() => {}}
  />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
