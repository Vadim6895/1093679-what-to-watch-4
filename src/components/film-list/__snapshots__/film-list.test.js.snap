// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmList render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
